File: 0392-is-subsequence/0392-is-subsequence.go
func isSubsequence(s string, t string) bool {
    if len(s) == 0 {
        return true
    }
    
    stack := []rune(s)
    for _, n := range t {
        if n == stack[0] {
            stack = stack[1:]
        }
        if len(stack) == 0 {
            return true
        }
    }
    return false
}

File: 0392-is-subsequence/0392-is-subsequence.c
bool isSubsequence(char * s, char * t){
    if (*s == '\0') 
        return true;
    
    for (; *t != '\0'; t++) {
        if (*s == *t)
            s++;
        if (*s == '\0')
            return true;
    }
    return false;
}